features/activity/retry_on_error/feature.ts

```diff
@@ -1,26 +1,46 @@
 import { Context } from '@temporalio/activity';
+import * as wf from '@temporalio/workflow';
 import { WorkflowFailedError } from '@temporalio/client';
 import { Feature } from '@temporalio/harness';
 import * as assert from 'assert';
-import { workflow } from './feature.workflow';
 
-export const feature = new Feature({
-  workflow,
-  activities: {
-    alwaysFailActivity,
-  },
-  checkResult: async (runner, handle) => {
-    await assert.rejects(runner.waitForRunResult(handle), (err) => {
-      assert.ok(
-        err instanceof WorkflowFailedError,
-        `expected WorkflowFailedError, got ${typeof err}, message: ${(err as any).message}`
-      );
-      assert.equal(err.cause?.cause?.message, 'activity attempt 5 failed');
-      return true;
-    });
+// Allow 4 retries with no backoff
+const activities = wf.proxyActivities<typeof activitiesImpl>({
+  startToCloseTimeout: '1 minute',
+  retry: {
+    // Retry immediately
+    initialInterval: '1 millisecond',
+    // Do not increase retry backoff each time
+    backoffCoefficient: 1,
+    // 5 total maximum attempts
+    maximumAttempts: 5,
   },
 });
 
-async function alwaysFailActivity() {
-  throw new Error(`activity attempt ${Context.current().info.attempt} failed`);
+export async function workflow(): Promise<void> {
+  // Execute activity
+  await activities.alwaysFailActivity();
 }
+
+const activitiesImpl = {
+  async alwaysFailActivity() {
+    throw new Error(`activity attempt ${Context.current().info.attempt} failed`);
+  },
+};
+
+export const feature =
```

Comment:
We can change the harness from `new Feature` to a static `Feature.create` or similar that returns a no-op feature in workflow context. This way, you still can have `proxyActivities<typeof feature.activities>`.

Reply:
You can't use anything imported from the harness in workflow code with this approach.
I tried to make the class workflow safe but encountered some issues with path aliases.
I think we could get rid of the condition eventually but it might require a bit more flexibility from the SDK bundler.

Reply:
> You can't use anything imported from the harness in workflow code with this approach.

I don't think you should use any non-static thing from the harness in the workflow (or would have to use globals instead like I had to in another test to access the runner/client from the activity). Utilities make sense, but things that are dependent on the current execution I hope are unnecessary inside the workflow.

Reply:
> I don't think you should use any non-static thing from the harness in the workflow

I am assuming agreement on this statement since nobody objected :-)

Reply:
Agreed